### PR TITLE
add check for unsafe strcpy()

### PR DESCRIPTION
**Pull request by maobourne**
add check for unsafe strcpy()

### CODE REVIEW

**Comment by drelaptop** [comment]
please check code style, change `TAB` to 4 spaces

File: cocos/scripting/lua-bindings/manual/network/lua_xml_http_request.cpp [hunk 1]
```diff
@@ -266,15 +266,18 @@ void LuaMinXmlHttpRequest::_setHttpRequestHeader()
         const char* second = it->second.c_str();
         size_t len = sizeof(char) * (strlen(first) + 3 + strlen(second));
         char* test = (char*) malloc(len);
-        memset(test, 0,len);
+        if (test != NULL) //should check failure of malloc or it could be an undefined behaviour
```
Comment (minggo) [inline cocos/scripting/lua-bindings/manual/network/lua_xml_http_request.cpp:269]: NULL -> nullptr

**Comment by minggo** [comment]
@drelaptop i think you can fix the tab issue based on this PR.

**Comment by drelaptop** [comment]
yep, I can do it later 3.17 release.

**Comment by drelaptop** [comment]
redo this change at #18885, for the author didn't response for a long time.

**Pull request closed without merging**